tests/nc-server/main.c

```diff
@@ -0,0 +1,106 @@
+#include <errno.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "events.h"
+#include "warnp.h"
+
+#include "simple_server.h"
+
+#define MAX_CONNECTIONS 2
+#define SHUTDOWN_AFTER 1
+
+struct nc_cookie {
+	FILE * out;
+};
+
+/* A client sent a message. */
+static int
+callback_snc_response(void * cookie, uint8_t * buf, size_t buflen)
+{
+	struct nc_cookie * C = cookie;
+
+	/* Write buffer to the previously-opened file. */
+	if (fwrite(buf, sizeof(uint8_t), buflen, C->out) != buflen) {
+		warnp("fwrite");
+		goto err0;
+	}
+
+	/* Success! */
+	return (0);
+
+err0:
+	/* Failure! */
+	return (-1);
+}
+
+int
+main(int argc, char ** argv)
+{
+	struct nc_cookie * C;
```

Comment:
Why not simply make this a structure on the stack; what do you gain by making this a pointer and the allocating the structure?